src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,373 @@
+            Else
+                Return Nothing
+            End If
+
+            If Not relevantSpan.IntersectsWith(textSpan.Start) Then
+                Return Nothing
+            End If
+
+            If token.SyntaxTree.OverlapsHiddenPosition(relevantSpan, cancellationToken) Then
+                Return Nothing
+            End If
+
+            Return relevantIfBlockOrIfStatement
+        End Function
+
+        Protected Overrides Function GetRootWithInvertIfStatement(workspace As Workspace, model As SemanticModel, ifStatement As SyntaxNode, cancellationToken As CancellationToken) As SyntaxNode
+            Dim result = UpdateSemanticModel(model, model.SyntaxTree.GetRoot().ReplaceNode(ifStatement, ifStatement.WithAdditionalAnnotations(s_ifNodeAnnotation)), cancellationToken)
+
+            Dim ifNode = FindIfNode(result.Root)
+
+            ' Complexify the top-most statement parenting this if-statement if necessary
+            Dim topMostExpression = ifNode.Ancestors().OfType(Of ExpressionSyntax).LastOrDefault()
+            If topMostExpression IsNot Nothing Then
+                Dim topMostStatement = topMostExpression.Ancestors().OfType(Of StatementSyntax).FirstOrDefault()
+                If topMostStatement IsNot Nothing Then
+                    Dim explicitTopMostStatement = Simplifier.Expand(topMostStatement, result.Model, workspace, cancellationToken:=cancellationToken)
+                    result = UpdateSemanticModel(result.Model, result.Root.ReplaceNode(topMostStatement, explicitTopMostStatement), cancellationToken)
+                    ifNode = FindIfNode(result.Root)
+                End If
+            End If
+
+            If (TypeOf ifNode Is SingleLineIfStatementSyntax) Then
+                model = InvertSingleLineIfStatement(workspace, DirectCast(ifNode, SingleLineIfStatementSyntax), result.Model, cancellationToken)
+            Else
+                model = InvertMultiLineIfBlock(DirectCast(ifNode, MultiLineIfBlockSyntax), result.Model, cancellationToken)
+            End If
+
+            ' Complexify the inverted if node.
+            result = (model, model.SyntaxTree.GetRoot())
+            Dim invertedIfNode = FindIfNode(result.Root)
+
+            Dim explicitInvertedIfNode = Simplifier.Expand(invertedIfNode, result.Model, workspace, cancellationToken:=cancellationToken)
+            result = UpdateSemanticModel(result.Model, result.Root.ReplaceNode(invertedIfNode, explicitInvertedIfNode), cancellationToken)
+
+            Return result.Root
+        End Function
+
+        Private Function UpdateSemanticModel(model As SemanticModel, root As SyntaxNode, cancellationToken As CancellationToken) As (Model As SemanticModel, Root As SyntaxNode)
+            Dim newModel = model.Compilation.ReplaceSyntaxTree(model.SyntaxTree, root.SyntaxTree).GetSemanticModel(root.SyntaxTree)
+            Return (newModel, newModel.SyntaxTree.GetRoot(cancellationToken))
```

Comment:
you do this update... and return the newModel.  but then you don't use it.  Can you explain what this is for?

Reply:
Ok.  i see you use this below.  can you add a comment explaining why we need to go through a set of semantic-model tranformation steps?